homeassistant/components/fritz/switch.py

```diff
@@ -522,61 +566,67 @@ async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+    def icon(self) -> str:
+        """Return device icon."""
+        return "mdi:router-wireless-settings"
+
+    async def async_process_update(self) -> None:
+        """Update device."""
+        if not self._mac:
+            return
+
+        wan_disable_info = await async_service_call_action(
+            self._router,
+            "X_AVM-DE_HostFilter",
+            "1",
+            "GetWANAccessByIP",
+            NewIPv4Address=self.ip_address,
         )
-        super().__init__(self._fritzbox_tools, device_friendly_name, switch_info)
 
-    async def _async_fetch_update(self) -> None:
-        """Update data."""
-        try:
-            status = await self.hass.async_add_executor_job(
-                self._fritzbox_tools.fritz_profiles[self.profile].get_state
-            )
-            _LOGGER.debug(
-                "Specific %s response: get_State()=%s",
-                SWITCH_TYPE_DEVICEPROFILE,
-                status,
-            )
-            if status == SWITCH_PROFILE_STATUS_OFF:
-                self._attr_is_on = False
-                self._is_available = True
-            elif status == SWITCH_PROFILE_STATUS_ON:
-                self._attr_is_on = True
-                self._is_available = True
-            else:
-                self._is_available = False
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.error("Could not get %s state", self.name, exc_info=True)
-            self._is_available = False
+        if wan_disable_info is None:
+            return
+
+        self._attr_is_on = not wan_disable_info["NewDisallow"]
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on switch."""
+        await self._async_handle_turn_on_off(turn_on=True)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off switch."""
+        await self._async_handle_turn_on_off(turn_on=False)
+
+    async def _async_handle_turn_on_off(self, turn_on: bool) -> bool:
+        """Handle switch state change request."""
+        await self._async_switch_on_off_executor(turn_on)
+        self._attr_is_on = turn_on
+        return True
 
     async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
-        """Handle profile switch."""
-        state = SWITCH_PROFILE_STATUS_ON if turn_on else SWITCH_PROFILE_STATUS_OFF
-        await self.hass.async_add_executor_job(
-            self._fritzbox_tools.fritz_profiles[self.profile].set_state, state
+        """Handle deflection switch."""
+        await async_service_call_action(
+            self._router,
+            "X_AVM-DE_HostFilter",
+            "1",
+            "DisallowWANAccessByIP",
+            NewIPv4Address=self.ip_address,
+            NewDisallow="0" if turn_on else "1",
         )
 
-    @property
-    def entity_registry_enabled_default(self) -> bool:
-        """Return if the entity should be enabled when first added to the entity registry."""
-        return False
-
```

Comment:
Please test carefully that other entities than Profiles are not disabled as well